chainermn/datasets/scatter_dataset.py

```diff
@@ -1,6 +1,41 @@
+import re
+import warnings
+
 import chainer.datasets
 import numpy
-import warnings
+
+
+class DataSizeError(RuntimeError):
+    def __init__(self, ds_size, pickled_size):
+        msg = """The dataset was too large to be scattered using MPI.
+
+        The length of the dataset is {} and it's size after being pickled
+        was {}. In the current MPI specification, the size cannot exceed
+        {}, which is so called 'INT_MAX'.
+
+        To solve this problem, please split the dataset into multiple
+        peaces and send/recv them separately.
+
+        Recommended sizes are indicated by ``slices()`` method.
```

Comment:
Could you delete this line because 'slices()' is not defined  in the current PR?.